components/mdx/index.tsx

```diff
@@ -37,24 +37,17 @@ const components = {
   Callout: ({ children }) => children,
   Bleed: ({ children }) => children,
   Codesandbox,
-  h4: ({ children, id, level }) => {
-    return (
-      <a href={`#${id}`} className="heading text-base mb-4 mt-4 tracking-tight">
-        <h4 id={id}>{children}</h4>
-      </a>
-    )
-  },
   Heading: ({ children, id, level }) => {
-    const Comp = level === 2 ? 'h2' : 'h3'
+    const headingStyle = {
+      2: 'text-3xl mb-6 mt-8',
+      3: 'text-xl mb-4 mt-6',
+      4: 'text-base mb-4 mt-4',
+    }
+
+    const clampedLevel = Math.min(Math.max(level, 2), 4)
+    const Comp = `h${clampedLevel}`
```

Comment:
TS is not recognizing this as a react element that accepts `children` and `id` props.